app/src/main/java/ch/epfl/sdp/appart/scrolling/card/CardAdapter.java

```diff
@@ -95,6 +106,21 @@ public void onBindViewHolder(@NonNull CardViewHolder holder, int position) {
         holder.priceTextView.setText(String.format("%d.-/mo", card.getPrice()));
         if (!card.hasVRTour())
             holder.vrAvailableImageView.setVisibility(View.INVISIBLE);
+
+        // Not the cleanest way of doing it, but it was difficult to do it differently with inheritance
+        Log.d("delete_ads", "This card is a UserCardAdapter");
+        if (isUserAd)
+            holder.deleteButton.setOnClickListener(v -> {
+                CompletableFuture<Boolean> deleteFuture = database.deleteAd(card.getAdId(), card.getId());
+                deleteFuture.thenAccept(b -> {
+                    if (b) {
+                        Intent intent = new Intent(context, UserAdsActivity.class);
+                        context.startActivity(intent);
```

Comment:
Are you re-creating the activity here?

Reply:
I'm starting it again to refresh it. Surely there may be a better way of doing it, I'll look into it.

Reply:
No it's fine, I just wanted to make sure I understood your intent!